### PR TITLE
LPS-180790 Scoping ObjectActionExecutor by objectDefinition by company

### PR DESCRIPTION
**Pull request by murilostodolni**

### CODE REVIEW

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/executor/FunctionObjectActionExecutorImpl.java [hunk 1]
```diff
@@ -71,6 +83,8 @@ protected void activate(Map<String, Object> properties) throws Exception {
 			ConfigurableUtil.createConfigurable(
 				FunctionObjectActionExecutorImplConfiguration.class,
 				properties);
+		_objectDefinitionNames = StringUtil.asList(
+			properties.get("objectDefinitionNames"));
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/executor/FunctionObjectActionExecutorImpl.java:87]: @murilostodolni sort `_objectDefinitionNames`

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/jaxrs/exception/mapper/ObjectActionExecutorKeyExceptionMapper.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	extends BaseExceptionMapper<ObjectActionExecutorKeyException> {
+
+	@Override
+	protected Problem getProblem(
+		ObjectActionExecutorKeyException objectActionExecutorKeyException) {
+
+		return new Problem(
+			Response.Status.BAD_REQUEST,
+			objectActionExecutorKeyException.getMessage());
+	}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/jaxrs/exception/mapper/ObjectActionExecutorKeyExceptionMapper.java:47]: @murilostodolni I believe if we intend to display the message for the client we should localize it by passing the property-key

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by murilostodolni** [comment]
ci:test:sf

File: workspaces/liferay-sample-workspace/client-extensions/liferay-sample-etc-spring-boot/client-extension.yaml [hunk 3]
```diff
@@ -22,6 +22,14 @@ liferay-sample-object-action-2:
     oAuth2ApplicationExternalReferenceCode: liferay-sample-oauth-application-user-agent
     resourcePath: /sample/object/action/2
     type: objectAction
+liferay-sample-object-action-3:
+    name: Liferay Sample Object Action 3
+    oAuth2ApplicationExternalReferenceCode: liferay-sample-oauth-application-user-agent
+    objectDefinitionNames:
+        - Object Definition 1 Name
+        - Object Definition 2 Name
```
Comment (MarinhoFeliphe) [inline workspaces/liferay-sample-workspace/client-extensions/liferay-sample-etc-spring-boot/client-extension.yaml:30]: @murilostodolni objectDefinitionName cannot has spaces, [see](https://github.com/liferay/liferay-portal/blob/7ba4cf99abeced68976df84ee8551414e022b02f/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java#L800-L813)

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

File: workspaces/liferay-sample-workspace/client-extensions/liferay-sample-etc-spring-boot/client-extension.yaml (hunk 3 above)
Comment (MarinhoFeliphe) [reply workspaces/liferay-sample-workspace/client-extensions/liferay-sample-etc-spring-boot/client-extension.yaml:30]: rename to ObjectDefinition1Name and ObjectDefinition2Name

**Comment by murilostodolni** [comment]
ci:test:sf

**Review by MarinhoFeliphe — APPROVED** [review]

**Comment by MarinhoFeliphe** [comment]
LGTM!! :tada:

**Comment by MarinhoFeliphe** [comment]
Very nice @murilostodolni !

**Comment by MarinhoFeliphe** [comment]
Resent here: https://github.com/drewbrokke/liferay-portal/pull/1419

**Pull request closed without merging**